python/tchannel/tornado/connection.py

```diff
@@ -518,6 +528,11 @@ def _stream(self, context, message_factory):
 
     @tornado.gen.coroutine
     def post_response(self, response):
+
+        # event: send_response
+        if self.tchannel:
+            self.tchannel.event_emitter.fire(EventType.send_response, response)
```

Comment:
So this is a good place to expand on that idea of EventEmitter conforming to the EventHook interface:
If EventEmitter conformed to the EventHook interface and acted as a composite EventHook of sorts, we could remove the knowledge of EventEmitter from TornadoConnection completely. It would only depend on a single EventHook-like object. In this place where you fire the event, it would just call "self.hook.send_response(..)".

When we injected an EventEmitter here, this send_response call would cause the EventEmitter to loop through all EventHooks and call send_response on them.

That'll would also remove the need of having this EventType enum which doesn't appear to have much value.